src/Esprima/JavascriptParser.cs

```diff
@@ -961,34 +982,31 @@ private BlockStatement ParsePropertyMethod(ref ParsedParameters parameters, out
         return body;
     }
 
-    private FunctionExpression ParsePropertyMethodFunction(bool isGenerator)
+    private FunctionExpression ParsePropertyMethodFunction(bool isGenerator, bool isAsync)
     {
         var node = CreateNode();
 
-        var previousAllowYield = _context.AllowYield;
-        _context.AllowYield = true;
-        var parameters = ParseFormalParameters();
-        var method = ParsePropertyMethod(ref parameters, out var hasStrictDirective);
-        _context.AllowYield = previousAllowYield;
-
-        return Finalize(node, new FunctionExpression(null, NodeList.From(ref parameters.Parameters), method, isGenerator, hasStrictDirective, false));
-    }
-
-    private FunctionExpression ParsePropertyMethodAsyncFunction(bool isGenerator)
-    {
-        var node = CreateNode();
+        var previousStrict = _context.Strict;
+        _context.Strict = false;
```

Comment:
This looks wrong.

For example
```js
({ m(x = 0644) { } })
```
compiles but
```js
'use strict';
({ m(x = 0644) { } })
```
doesn't.

On top of that, the entire body of classes is in strict mode explicitly IIRC.